autolab/Autolab
# Check for handin file type when uploading file to assignment

## Description

Developed as part of the UI Audit, in order to make the expected file type communicated to students more clearly.
- Adds simple file type checking logic when uploading a file to Autolab, if the handin file has a "." (e.g. ".c", ".py"), otherwise logic checking won't occur (if for example specified file type is just "README")
- If the file type specified in assessment settings doesn't match the uploaded file / archive, display a visual warning to the user, but don't block uploading of the file as to not break any current features

Closes #951 

## How Has This Been Tested?




- Go to "Edit Assessment," and then "handin filename," name as "hello.c," and save settings
- Go to handin area and see that the expected file type shows up under the upload area, and that when you upload a c file, that no warning appears

<img width="714" alt="Screen Shot 2022-11-09 at 14 29 18" src="https://user-images.githubusercontent.com/25730111/200931222-003079b3-ae35-46cc-bca6-fb9b162fc2a8.png">
- try uploading a different file type and see that an error occurs

<img width="663" alt="Screen Shot 2022-11-09 at 14 28 48" src="https://user-images.githubusercontent.com/25730111/200931218-b83063f2-01f2-4530-8034-ba6aa285cb4e.png">
- upload a file with multiple periods but ends in .c, see that error disappears

<img width="683" alt="Screen Shot 2022-11-09 at 14 29 29" src="https://user-images.githubusercontent.com/25730111/200931225-ef0adfcd-5da7-4289-85fc-33b3dba44d7f.png">
- Click on github and see that specified file type disappears

<img width="725" alt="Screen Shot 2022-11-09 at 15 03 24" src="https://user-images.githubusercontent.com/25730111/200931227-2c3d6bfd-661d-4195-aa5b-6f5b2cc3771e.png">

- also check .tar files, and editing assessment handin filename to have no filetype

## Types of changes


## Checklist:
